### PR TITLE
[WIP] Adding GraphQL endpoints with tests 

### PR DESCRIPTION
**Pull request by tgrochowski**
| Q               | A
| --------------- | -----
| Branch?         |  master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Following PR is in progress for developing experimental GraphQL APIPlatform feature along with Behat tests covering newly added GraphQL mutations and queries.

The goal for first iteration of published work was to create working login mutation and complete simple order with ability to browse through taxons and products with basic filtering.

Ultimately this GraphQL endpoints shall be used with VueStoreFront 2, frontend shop implementation.

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Command/Cart/ApplyCouponToCart.php [hunk 1]
```diff
@@ -24,14 +24,15 @@ class ApplyCouponToCart implements OrderTokenValueAwareInterface
     /** @var string|null */
     public $couponCode;
 
-    public function __construct(?string $couponCode)
+    public function __construct(?string $couponCode,?string $orderTokenValue)
```
Comment (plozmun) [inline src/Sylius/Bundle/ApiBundle/Command/Cart/ApplyCouponToCart.php:27]:
```suggestion
    public function __construct(?string $couponCode, ?string $orderTokenValue)
```
Missing space

File: src/Sylius/Bundle/ApiBundle/Command/Cart/ApplyCouponToCart.php [hunk 2]
```diff
@@ -24,14 +24,15 @@ class ApplyCouponToCart implements OrderTokenValueAwareInterface
...
     {
         $this->couponCode = $couponCode;
+        $this->orderTokenValue = $orderTokenValue;
     }
 
     public static function createFromData(string $orderTokenValue, ?string $couponCode): self
     {
-        $command = new self($couponCode);
+        $command = new self($couponCode,$orderTokenValue);
```
Comment (plozmun) [inline src/Sylius/Bundle/ApiBundle/Command/Cart/ApplyCouponToCart.php:35]:
```suggestion
        $command = new self($couponCode, $orderTokenValue);
```
Same as above :)

**Comment by Arminek** [comment]
Nice! Could you split this pr into some smaller chunks? I would split it by features. This will speed up the process of reviewing.

**Comment by tgrochowski** [comment]
For now this PR will be put on hold as it wont fit in upcoming release cycles, it also modifies strict parts of Sylius core that might cause issues in experimental code so in that case we have decided to postpone this and move that logic to separate Sylius plugin. Plugin will be shared with community after polishing it and testing properly.

**Pull request closed without merging**